eureka-core/src/main/java/com/netflix/eureka/util/AwsAsgUtil.java

```diff
@@ -313,6 +385,29 @@ public void run() {
         return asgNames;
     }
 
+    /**
+     * Get the AWS account id where an ASG is created.
+     *
+     * @param asgName
+     *            - The name of the ASG
+     * @return the account id
+     */
+    private String getASGAccount(String asgName) {
+        Applications apps = PeerAwareInstanceRegistry.getInstance().getApplications(false);
+
+        for (Application app : apps.getRegisteredApplications()) {
+            for (InstanceInfo instanceInfo : app.getInstances()) {
+                String thisAsgName = instanceInfo.getASGName();
+                if (thisAsgName != null && thisAsgName.equals(asgName)) {
+                    return ((AmazonInfo) instanceInfo.getDataCenterInfo()).get(MetaDataKey.accountId);
```

Comment:
The accountId metadata field was added to eureka-client relatively recently, so for backwards compatibility, it is possible in existing deployments to have older clients that do not register this field. This will return null here and cause an NPE upstream. Can you please add a null check here and let it fall through instead?